src/Fixer/ControlStructure/EmptyLoopBodyFixer.php

```diff
@@ -0,0 +1,146 @@
+                    [
+                        'style' => 'semicolon',
+                    ]
+                ),
+            ]
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * Must run before BracesFixer, NoExtraBlankLinesFixer.
+     * Must run after NoEmptyStatementFixer.
+     */
+    public function getPriority(): int
+    {
+        return 39;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isCandidate(Tokens $tokens): bool
+    {
+        return $tokens->isAnyTokenKindsFound($this->tokenLoopKinds);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function applyFix(\SplFileInfo $file, Tokens $tokens): void
+    {
+        if (self::STYLE_BRACES === $this->configuration['style']) {
+            $analyzer = new TokensAnalyzer($tokens);
+            $fixLoop = static function (int $index, int $endIndex) use ($tokens, $analyzer): void {
+                if ($tokens[$index]->isGivenKind(T_WHILE) && $analyzer->isWhilePartOfDoWhile($index)) {
+                    return;
+                }
+
+                $semiColonIndex = $tokens->getNextMeaningfulToken($endIndex);
+
+                if (!$tokens[$semiColonIndex]->equals(';')) {
+                    return;
+                }
+
+                $tokens[$semiColonIndex] = new Token('{');
+                $tokens->insertAt($semiColonIndex + 1, new Token('}'));
+            };
+        } else {
+            $fixLoop = static function (int $index, int $endIndex) use ($tokens): void {
+                $braceOpenIndex = $tokens->getNextMeaningfulToken($endIndex);
+
+                if (!$tokens[$braceOpenIndex]->equals('{')) {
+                    return;
+                }
+
+                $braceCloseIndex = $tokens->getNextMeaningfulToken($braceOpenIndex);
+
+                if (!$tokens[$braceCloseIndex]->equals('}')) {
+                    return;
+                }
+
+                $tokens[$braceOpenIndex] = new Token(';');
+                $tokens->clearTokenAndMergeSurroundingWhitespace($braceCloseIndex);
+            };
+        }
+
+        for ($index = $tokens->count() - 1; $index > 0; --$index) {
+            if ($tokens[$index]->isGivenKind($this->tokenLoopKinds)) {
+                $endIndex = $tokens->getNextTokenOfKind($index, ['(']); // proceed to open '('
+                $endIndex = $tokens->findBlockEnd(Tokens::BLOCK_TYPE_PARENTHESIS_BRACE, $endIndex); // proceed to close ')'
+                $fixLoop($index, $endIndex); // fix loop if needs fixing
+            }
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function createConfigurationDefinition(): FixerConfigurationResolverInterface
+    {
+        return new FixerConfigurationResolver([
+            (new FixerOptionBuilder('style', 'Style of empty loop-bodies.'))
+                ->setAllowedTypes(['string'])
+                ->setAllowedValues([self::STYLE_BRACES, self::STYLE_SEMICOLON])
+                ->setDefault(self::STYLE_BRACES)
```

Comment:
I think `;` is better to express that the empty loop is intentional, I would make it the default.

Reply:
Not enough to assume they want to follow the `braces` style IMO. /cc @nicolas-grekas maybe?

Reply:
better use `{}` I suppose since it's a bit more readable/explicit? I've no strong opinion on the topic.

Reply:
Ok, let's keep `braces` as default value then.

Reply:
I would love to put one more argument. real example from our own codebase:
```diff
        for (++$index; $tokens[$index]->isWhitespace(" \t") || $tokens[$index]->isComment(); ++$index);
        for (++$index; $tokens[$index]->isWhitespace(" \t") || $tokens[$index]->isComment(); ++$index) {
        }
```

- version using `{}` - oh, we had a loop that was doing sth one day, and now we no longer execute anything inside, so the whole loop can be dropped! awesome, dead code detected, removing ;) ## ups, I broke the code
- version using `;` - ah, we are moving the index, the code is needed.

(so basically what @julienfalque said)

-----

I strongly suggest to use `;` by default.
@nicolas-grekas , can you reconsider your opinion?

@julienfalque , @SpacePossum  - I suggest to make fixer default `;` whatever gonna be decision of for Symfony ruleset

Reply:
(ref) https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/5745#issuecomment-889763849
> I've changed the default and removed the rule from the Symfony set. This way the talk about (default) value in sets can be done in new PR's with representation of the community of the set, while not blocking this PR from proceeding.

@nicolas-grekas , don't hesitate to create new ticket to update Symfony set